### ISSUE TITLE
OpenGL renderer sketches don't run

### ISSUE THREAD

**Issue by SmantikVJ** [processing/processing#5866]
I have some problems with Processing in my new Ubuntu 18.04 64 bits ARM-A57 Nvidia Jetson Nano SBC with Nvidia Maxwell 128 core GPU.
After installing the program I wasn't able to execute it due of problems with Oracle Java 8 version that the processing ARM version has embedded.After looking for a valid solution, an user (simon-ritter) told me in a post that I could replace the java folder that processing includes with the Oracle Java 8 version with the untared _zulu8.38.0.162-ca-jdk1.8.0_212-linux_aarch64.tar.gz_ and change the name of the folder to "java".It was a valid solution and now I can execute Processing and run its sketches with no problem, if I use the default renderer. The problem appears when I try to run a sketch that uses the P2D or the P3D, which are OpenGL renderers both of them. Then, Processing shows the following error message and create an error .log file (attached). Suggested solution of "_try "ulimit -c unlimited_" before starting Java again"  doesn't work.Is there any solution for this issue?
Thank you.
[hs_err_pid22731.log](https://github.com/processing/processing/files/3232060/hs_err_pid22731.log)

**Comment by SmantikVJ** [comment]
When I check the OpenGL version, I can see this:
_glxinfo | grep "OpenGL version"
OpenGL version string: 4.6.0 NVIDIA 32.1.0_

**Comment by dzaima** [comment]
Processing 3 doesn't officially support OpenJDK which is probably the cause, but there has been work on an [OpenJDK 11 version](https://github.com/sampottinger/processing). The whole ARM compatibility issue should be solved before worrying about OpenGL.

**Comment by SmantikVJ** [comment]
Ok. So, I'll have to go back to my Raspberry PI 3B+ SBC that has no problems with Java, neither with OpenGL P2D and P3D renderers, neither with the Contribution Manager for downloading libraries, tools, etc....
I'll place the nvidia Jetson Nano in the refrigerator.

**Comment by SmantikVJ** [comment]
Incredible, I have Processing sketches running with OpenGL renderers. Although when I close them you can see some errors in the console, the sketch runs now.
I have read the post of Diablo in https://devtalk.nvidia.com/default/topic/1051204/jetson-nano/how-detected-tegra-gl-core-correctly-in-cmake-/post/5335756/?offset=5#5347306 and he gave me an idea. I changed the /usr/lib/aarch64-linux-gnu/libGLX.so.0 problematic file for the /usr/lib/aarch64-linux-gnu/tegra/libGLX_nvidia.so.0 file. Then I rename this file to libGLX.so.0 and Processing now can run OpenGL rendered sketches.
https://drive.google.com/open?id=1h8bDiVyrTlOazdEIJeJS4GI5lvw9Teew
The problem seems to be that Processing when uses OpenGL doesn't link to tegra lib and it goes to libGLX.so.0 I'm not a computer technician, I can be wrong but it works.
:D

This are the final errors that appear when I close the sketch:
https://drive.google.com/open?id=1CnzQ0eBjvmjBqiRu8IZXd4A8Zcq9S1-A

**Comment by dzaima** [comment]
@SmantikVJ I believe the "errors" upon closing aren't really supposed to be interpreted as errors, they're just in the stderr stream (I get them too and i haven't messed with java versions).  
@gouessej sorry for pinging if this isn't anything to do with JOGL (hoping for the best since you were involved here in another issue here about it), but maybe the different shared library files are something fixable on JOGLs side?

**Comment by gouessej** [comment]
@dzaima You should ask this question to @xranby as he uses the Raspberry Pi more often than me: http://forum.jogamp.org/What-is-the-status-of-font-rendering-with-JOGL-tp4039805p4039812.html He talked about removing /opt/vc/lib

I'll leave Github one day, consider asking JOGL specific questions on our official forum.

**Comment by benfry** [comment]
Closing, as this is well outside our control…
